### ISSUE TITLE
ERROR: Module "features" does not exist

### ISSUE THREAD

**Issue by miaodi** [mesonbuild/meson#12572]
I am trying to build numpy via from source code. However, when I do

`meson setup ..`

I got the following:

```The Meson build system
Version: 1.3.99
Source dir: /SCRATCH/dimiao/test_space/numpy-1.26.2
Build dir: /SCRATCH/dimiao/test_space/numpy-1.26.2/build
Build type: native build
Project name: NumPy
Project version: 1.26.2
C compiler for the host machine: /SCRATCH/dimiao/test_space/installed/gcc-13/bin/gcc (gcc 12.3.0 "gcc (GCC) 12.3.0")
C linker for the host machine: /SCRATCH/dimiao/test_space/installed/gcc-13/bin/gcc ld.bfd 2.41.50.20231129
C++ compiler for the host machine: /SCRATCH/dimiao/test_space/installed/gcc-13/bin/g++ (gcc 12.3.0 "g++ (GCC) 12.3.0")
C++ linker for the host machine: /SCRATCH/dimiao/test_space/installed/gcc-13/bin/g++ ld.bfd 2.41.50.20231129
Cython compiler for the host machine: cython (cython 3.0.6)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 found: YES (/SCRATCH/dimiao/test_space/.venv/bin/python3)
Found pkg-config: YES (/usr/bin/pkg-config) 0.27.1
Run-time dependency python found: YES 3.12
Has header "Python.h" with dependency python-3.12: YES 
Compiler for C supports arguments -fno-strict-aliasing: YES 

../meson_cpu/x86/meson.build:2:15: ERROR: Module "features" does not exist```

I tried both meson from pip install (1.3.0) and meson from source code (1.3.99).

**Comment by eli-schwartz** [comment]
See #11307 -- note that numpy is writing this functionality (and also a blas dependency) and also, as I understand it, field-testing it to make sure it handles everything it needs to handle, which is why @seiko2plus and @rgommers aren't quite ready to land it. This means that numpy will only build using their vendored fork for the moment.

**Comment by rgommers** [comment]
Indeed, using `meson setup ...` directly cannot work - although you can use `python vendored-meson/meson/meson.py setup ...` instead if you for some reason really wish to avoid using `pip`/`build`.

> also, as I understand it, field-testing it to make sure it handles everything it needs to handle, which is why @seiko2plus and @rgommers aren't quite ready to land it.

Indeed. I'm about ready for BLAS support, just one more open bug report to handle and then need to sit down for a day to update everything in my PR.

**Comment by rubyFeedback** [comment]
I just ran into the same problem trying to build numpy from source:

    Program python3 found: YES (/usr/bin/python)
    Found pkg-config: YES (/usr/bin/pkg-config) 1.8.1
    Run-time dependency python found: YES 3.11
    Has header "Python.h" with dependency python-3.11: YES 
    Compiler for C supports arguments -fno-strict-aliasing: YES 

    meson_cpu/x86/meson.build:2:15: ERROR: Module "features" does not exist

Absolutely no idea how to resolve this.

Also by the way, googling for module features is very hard. Whoever named it
that way evidently was thinking "hey, nobody will ever run into this problem".
"features" is such a common word ... google search is really bad at it. This
was the first issue that popped up on google search and it has not helped 
solve the issue, so now I have to hop to another search result ...

**Comment by rgommers** [comment]
@rubyFeedback you are probably running `meson setup` in the root of the numpy repo or an unpacked sdist. That is something you cannot do. Use `pip install`, `python -m build --wheel` or similar. If you must run `meson` directly, use `vendored-meson/meson/meson.py` inside the numpy source tree.
